src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -111,11 +116,23 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
-          this.onError.bind(this),
-          this.props.notifications!,
-          this.props.onSelectedDataSources
-        );
+        // return [] to customer when no default data source
+        if (!defaultDataSource && this.props.hideLocalCluster) {
+          this.props.onSelectedDataSources?.([]);
+        }
+        const filteredDefaultDataSource = this.getFilteredDefaultDataSource(defaultDataSource);
+        this.setState({
+          defaultDataSource: filteredDefaultDataSource,
+        });
+        const handleDataSourceViewErrorParams = {
+          changeState: this.onError.bind(this),
+          notifications: this.props.notifications!,
+          failedDataSourceId: optionId,
+          defaultDataSource: filteredDefaultDataSource,
+          callback: this.handleSwitchDefaultDatasource.bind(this, this.state.defaultDataSource),
```

Comment:
can we use filteredDefaultDataSource here?

Reply:
updated